nav2_behavior_tree/plugins/action/compute_path_to_pose_action.cpp

```diff
@@ -42,15 +42,15 @@ BT::NodeStatus ComputePathToPoseAction::on_success()
   setOutput("path", result_.result->path);
   // Set empty error code, action was successful
   result_.result->error_code = nav2_msgs::action::ComputePathToPose::Goal::NONE;
-  setOutput("compute_path_to_pose_error_code", result_.result->error_code);
+  config().blackboard->set<int>("compute_path_to_pose_error_code", result_.result->error_code);
   return BT::NodeStatus::SUCCESS;
 }
 
 BT::NodeStatus ComputePathToPoseAction::on_aborted()
 {
   nav_msgs::msg::Path empty_path;
   setOutput("path", empty_path);
-  setOutput("compute_path_to_pose_error_code", result_.result->error_code);
+  config().blackboard->set<int>("compute_path_to_pose_error_code", result_.result->error_code);
```

Comment:
The output port was the right place, not the blackboard. The output port can be set to a blackboard value using `{my_var_name}` (so `{compute_path_to_pose_error_code}`), but then allows someone to remap this name if they like, since its conceivable that there are multiple instances of these action nodes in a BT so you might want to differentiate them. 

So the use of this BT node would be `<ComputePathToPose error_code_id={compute_path_to_pose_error_code}/>`

Note that I simplified `error_code`, I think it would actually be wise to set the output port for all of these actions with failure codes to a generic name like that for brevity of the BT. As a result, then all of the actual blackboard key names must be variables/parameters (e.g. `compute_path_to_pose_error_code`). 

For example, for the other "special" output/input values of the BT, we have parameters for the node that have to match the BT in use (https://github.com/ros-planning/navigation2/blob/main/nav2_bt_navigator/src/navigators/navigate_to_pose.cpp#L33-L43). That way you can change the goal ID name in the BT `{goal}` to `{my_goal}` for whatever reason and then can tell BT navigator what key it should look for for the goal without naively assuming its hardcoded to `goal`. 

So 
- There needs to be a parameterization of BT navigator for the error code blackboard names to use to look for error codes, and that used to set it like you do in the file below this one 
- The use of the error code must be set in the BT at the output port so that they match up, for all the BTs
- Rename the `setOutput` / `outputPort` (after adding back in) to `error_code_id` for shortening, since that's not the actual name of the variable being used, we can create a common name foundation for this in all the BT nodes

The question kind of remains though in the BT navigator how to handle if we have multiple key names for a compute path action to get them for populating the final message. I suppose we could make the set of follow / compute path / etc error code parameter field vectors? 

---

Or actually, maybe we can make a global const name for the names of these error code port names so they can still be modified but will all be the same from the BT navigator + BT nodes. `const std::string path_error_code_name = "my_code_name"`, and use that variable everywhere. It wouldn't be run-time configurable, only compile-time.

Reply:
So in line with the other comment, there would be no need to specify which error code IDs belong to which server, we could just take in the vector of `error_code_ids` and populate them to `0` (which we should def to a variable of name `NONE`) instead. I actually don't see anywhere in the code of BT navigator where we specifically need to know the name of a given servers error code id. Just as long as they're all in `error_code_ids` and specified in the BT output ports have matching sets.